tests/driver_pca95xx/main.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (C) 2014,2017 Freie Universität Berlin
+ *               2018 Matthew Blue <matthew.blue.neuro@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Test application PCA95xx GPIO expanders
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *              Matthew Blue <matthew.blue.neuro@gmail.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "irq.h"
+#include "pca95xx.h"
+#include "pca95xx_params.h"
+#include "shell.h"
+
+static pca95xx_t pca95xx_dev = { .params.i2c   = PCA95XX_PARAM_I2C,
+                                 .params.addr  = PCA95XX_PARAM_ADDR,
+                                 .params.flags = PCA95XX_PARAM_FLAGS };
+
+static pca95xx_int_t pca95xx_int_dev = { .params.i2c     = PCA95XX_PARAM_I2C,
+                                         .params.addr    = PCA95XX_PARAM_ADDR,
+                                         .params.int_pin = PCA95XX_PARAM_INT_PIN };
+
+static void cb(void *arg)
+{
+    (void)arg;
+    puts("PCA95xx pin change interrupt");
+}
+
+static int init_pin(int argc, char **argv)
+{
+    if (argc < 2) {
+        printf("usage: %s <pin>\n", argv[0]);
+        return 1;
+    }
+
+    pca95xx_dev.params.pin = atoi(argv[1]);
```

Comment:
normally, the params attribute correspond to initialization parameters and should not updated afterward. Would it make sense to introduce a new function in the API to initialize a specific pin instead of reinitializing the whole device?